### PR TITLE
[keyboard]Add Graystudio Space65 R3

### PR DESCRIPTION
**Pull request by edwardslau**
## Description

Add Space65R3 
## Types of Changes

- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by edwardslau** [comment]
@drashna Hi, can you give me some help? I have trouble finding the error. The files compile perfectly on my end.

File: keyboards/gray_studio/space65r3/config.h [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+/*== or choose animations ==*/
+  // #define RGBLIGHT_EFFECT_BREATHING
+  // #define RGBLIGHT_EFFECT_RAINBOW_MOOD
+  // #define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+  // #define RGBLIGHT_EFFECT_SNAKE
+  // #define RGBLIGHT_EFFECT_KNIGHT
+  // #define RGBLIGHT_EFFECT_CHRISTMAS
+  // #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+  // #define RGBLIGHT_EFFECT_RGB_TEST
+  // #define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/config.h:58]:
```suggestion
```

File: keyboards/gray_studio/space65r3/info.json [hunk 2]
```diff
@@ -0,0 +1,239 @@
+{
+    "keyboard_name": "Gray Studio Space65R3",
+    "url": "",
+    "maintainer": "qmk",
+    "width": 16,
+    "height": 5,
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/info.json:6]:
```suggestion
```

File: keyboards/gray_studio/space65r3/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,76 @@
+
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/keymaps/default/keymap.c:2]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/gray_studio/space65r3/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),                     KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT(
+        RESET,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, _______,
+        _______,          RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, _______, _______, _______, _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______, _______,          _______, _______,
+        _______, _______, _______,                            _______,                   _______, _______,                   _______, _______, _______
+    ),
+    [2] = LAYOUT(
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/keymaps/default/keymap.c:19]: please remove the empty layers.

File: keyboards/gray_studio/space65r3/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,76 @@
+
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/keymaps/via/keymap.c:1]: same here

File: keyboards/gray_studio/space65r3/mcuconf.h [hunk 6]
```diff
@@ -0,0 +1,27 @@
+/* Copyright 2020 QMK
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/mcuconf.h:1]: this file can be removed.

File: keyboards/gray_studio/space65r3/halconf.h [hunk 7]
```diff
@@ -0,0 +1,25 @@
+/* Copyright 2020 QMK
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/halconf.h:1]: this file should be removed.

File: keyboards/gray_studio/space65r3/rules.mk [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/rules.mk:21]: Please remove the `SLEEP_LED_ENABLE` setting, as it's not really used, and remove the comments for both it and NKRO.  
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/gray_studio/space65r3/rules.mk [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/rules.mk:28]:
```suggestion
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/gray_studio/space65r3/space65r3.c [hunk 10]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/space65r3.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/gray_studio/space65r3/space65r3.h [hunk 11]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/space65r3.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by edwardslau** [comment]
@drashna I am still having problems with the lighting. Can you help me out on this one?

**Comment by drashna** [comment]
> @drashna I am still having problems with the lighting. Can you help me out on this one?

It looks like the only issues are the mechmerlin and spidey3 keymaps are not compiling.  These are likely community layouts, so they can be safely ignored. The other keymaps compile file.

Eg: 
https://github.com/qmk/qmk_firmware/pull/18909/checks?check_run_id=9588958570

File: keyboards/gray_studio/space65r3/config.h [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 1
+
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/config.h:184]:
```suggestion
```

File: keyboards/gray_studio/space65r3/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+        _______, _______, _______,                            _______,                   _______, _______,                   _______, _______, _______
+    ),
+    [3] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+        _______, _______, _______,                            _______,                   _______, _______,                   _______, _______, _______
+    ),
+    [4] = LAYOUT(
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/keymaps/via/keymap.c:34]: The default is 4 layers.  If it supports more than that, the dynamic keymap define should be changed to support 8.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/gray_studio/space65r3/keymaps/via/keymap.c (hunk 13 above)
Comment (sigprof) [reply keyboards/gray_studio/space65r3/keymaps/via/keymap.c:34]: Are those empty layers actually required after #17695 (already in master)?

**Comment by edwardslau** [comment]
> > @drashna I am still having problems with the lighting. Can you help me out on this one?
> 
> It looks like the only issues are the mechmerlin and spidey3 keymaps are not compiling. These are likely community layouts, so they can be safely ignored. The other keymaps compile file.
> 
> Eg: https://github.com/qmk/qmk_firmware/pull/18909/checks?check_run_id=9588958570

Should I just simply ignore it? If so I have fixed other issues.

**Comment by drashna** [comment]
> > > @drashna I am still having problems with the lighting. Can you help me out on this one?
> > 
> > 
> > It looks like the only issues are the mechmerlin and spidey3 keymaps are not compiling. These are likely community layouts, so they can be safely ignored. The other keymaps compile file.
> > Eg: https://github.com/qmk/qmk_firmware/pull/18909/checks?check_run_id=9588958570
> 
> Should I just simply ignore it? If so I have fixed other issues.

Yes. 

However, there are some other changes needed now.

File: keyboards/gray_studio/space65r3/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,52 @@
+    // Copyright 2022 Yizhen Liu (@edwardslau)
+// SPDX-License-Identifier: GPL-2.0
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_DEL,
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/keymaps/default/keymap.c:7]: This has been changed: 
```suggestion
        QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_DEL,
```

File: keyboards/gray_studio/space65r3/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 Yizhen Liu (@edwardslau)
+// SPDX-License-Identifier: GPL-2.0
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_DEL,
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/keymaps/via/keymap.c:7]:
```suggestion
        QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_DEL,
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by edwardslau** [comment]
> > > > @drashna I am still having problems with the lighting. Can you help me out on this one?
> > > 
> > > 
> > > It looks like the only issues are the mechmerlin and spidey3 keymaps are not compiling. These are likely community layouts, so they can be safely ignored. The other keymaps compile file.
> > > Eg: https://github.com/qmk/qmk_firmware/pull/18909/checks?check_run_id=9588958570
> > 
> > 
> > Should I just simply ignore it? If so I have fixed other issues.
> 
> Yes.
> 
> However, there are some other changes needed now.

Thanks a lot! I have committed to the changes.

File: keyboards/gray_studio/space65r3/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),                     KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT(
+        RESET,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, _______,
+        _______,          RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, _______, _______, _______, _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______, _______,          _______, _______,
+        _______, _______, _______,                            _______,                   _______, _______,                   _______, _______, _______
+    ),
+      [2] = LAYOUT_all(
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/keymaps/default/keymap.c:20]:
```suggestion
      [2] = LAYOUT(
```

Though, these layers aren't used, and could be removed.

File: keyboards/gray_studio/space65r3/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            _______,            _______,
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            _______,
+    _______,            _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,
+    _______,            _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            _______,
+    _______,            _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            _______,  _______,  _______,
+    _______,  _______,            _______,                                _______,                      _______,  _______,            _______,  _______,  _______
+  ),
+ 
+  [3] = LAYOUT_all(
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/keymaps/default/keymap.c:29]:
```suggestion
  [3] = LAYOUT(
```

File: keyboards/gray_studio/space65r3/keymaps/via/keymap.c [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+        _______, _______, _______,                            _______,                   _______, _______,                   _______, _______, _______
+    ),
+    [3] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
+        _______, _______, _______,                            _______,                   _______, _______,                   _______, _______, _______
+    )，
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/keymaps/via/keymap.c:33]:
```suggestion
    )
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/gray_studio/space65r3/config.h [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

**Review by drashna — APPROVED** [review]

File: keyboards/gray_studio/space65r3/keymaps/default/keymap.c [hunk 20]
```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 Yizhen Liu (@edwardslau)
+// SPDX-License-Identifier: GPL-2.0
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_DEL,
```
Comment (drashna) [inline keyboards/gray_studio/space65r3/keymaps/default/keymap.c:7]: CI says that there are too many elements when compiling?

Comment (edwardslau) [reply keyboards/gray_studio/space65r3/keymaps/default/keymap.c:7]: @drashna I have updated the keymap

**Review by tzarc — APPROVED** [review]

**Pull request merged**